packages/docusaurus/src/server/plugins/index.ts

```diff
@@ -5,21 +5,51 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import {DocusaurusConfig} from '../config';
+import {LoadContext, Props} from '..';
+import {RouteConfig} from '../routes';
+
 import fs from 'fs-extra';
 import importFresh from 'import-fresh';
 import path from 'path';
 import {generate} from '@docusaurus/utils';
-import {LoadContext} from '..';
+
+export interface Plugin<T> {
+  name: string;
+  loadContent?(): T;
+  contentLoaded?({
+    content: T,
+    actions: DocusaurusPluginContentLoadedActions,
+  }): void;
+  postBuild?(props: Props): void;
+  postStart?(props: Props): void;
+  configureWebpack?(config: DocusaurusConfig, isServer: boolean);
```

Comment:
not docusaurus config but actual  webpack.configuration. result is also webpack config